src/components/WebCLI/WebCLI.tsx

```diff
@@ -127,22 +127,53 @@ const WebCLI = ({
     const conn = new Connection({
       address: wsAddress,
       onopen: () => {
-        // Unused handler.
+        // Close this websocket connection if it was in CONNECTING state
+        // while a new websocket connection was established.
+        if (conn.websocket !== connection.current?.websocket) {
+          conn.disconnect();
+        }
       },
       onclose: () => {
         // Unused handler.
       },
+      onerror: (error) => {
+        setInlineError(
+          InlineErrors.CONNECTION,
+          typeof error === "string" ? error : "Unknown error.",
+        );
```

Comment:
To prevent the error from the old websocket closing while connecting we can check that the error is related to the current connection.

```suggestion
        // Only display errors if they're related to the current websocket connection.
        if (conn.websocket === connection.current?.websocket) {
          setInlineError(
            InlineErrors.CONNECTION,
            typeof error === "string" ? error : "Unknown error.",
          );
        }
```

Reply:
Thanks for the review! This works. I've changed the code slightly so that I don't expose the `wesocket` entirely, but just compare if 2 `Connections` are equal with a method within the class.